### ISSUE TITLE
Image pulls or runs result in the process hanging indefinitely with spinning icons even after updating

### ISSUE THREAD

**Issue by ylluminate** [docker/for-mac#6799]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 64F7076D-1789-47DD-922B-37C245F5E296/20230411042337

### Expected behavior

Docker Desktop should successfully download and run images without hanging indefinitely.

### Actual behavior

When attempting to download an image using `pull` or `run`, the process hangs indefinitely with spinning icons. A blue line appears in the image list name of the image, but it never changes or progresses.

### Information
  - Is it reproducible? Yes
  - Is the problem new? Yes
  - Did the problem appear with an update? Not that I'm aware of
  - macOS Version: 12.6.4
  - Apple chip: M1 Max
  - Docker Desktop Version: 4.18.0

### Output of `/Applications/Docker.app/Contents/MacOS/com.docker.diagnose check`

```
$ /Applications/Docker.app/Contents/MacOS/com.docker.diagnose check
[2023-04-11T04:18:33.063494000Z][com.docker.diagnose][I] set path configuration to OnHost
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0013: is the $PATH ok?
[PASS] DD0003: is the Docker CLI working?
[PASS] DD0038: is the connection to Docker working?
[PASS] DD0014: are the backend processes running?
[PASS] DD0007: is the backend responding?
[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0033: does the host have Internet access?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0032: do Docker networks overlap with host IPs?
No fatal errors detected.
```

### Steps to reproduce the behavior

  1. Open Docker Desktop on macOS
  2. Attempt to download an image using `pull` or `run`
  3. Observe that the process hangs indefinitely, and a blue line appears in the image list name of the image without changing or progressing.

**Comment by boehsermoe** [comment]
Same issue here!

**Comment by bsousaa** [comment]
Can you confirm if your issue happens with the latest Docker Desktop version? You can get the latest at https://docs.docker.com/get-docker/

What image specifically are you pulling?

**Comment by boehsermoe** [comment]
@bsousaa Yes I can confirm it. I did also try to downgrade to v4.18, but also the same problem. 

I also tried restart the mac (v13.4.1) 

I tested with hello-world image and there is the same problem.

`docker pull hello-world       
Using default tag: latest
latest: Pulling from library/hello-world
70f5ac315c5a: Waiting 
`

**Comment by boehsermoe** [comment]
Ok, after Update again to latest version 4.20.1 it working again! Maybe the reinstall did fix it.

**Comment by bsousaa** [comment]
@ylluminate can you confirm if the issue persists with latest 4.20.1?

**Comment by ylluminate** [comment]
Unfortunately I tried to pull Alpine and seem to still have the issue. Here's a short screencast of the GUI result:
![screencast](https://p153.p0.n0.cdn.getcloudapp.com/items/5zuOGRA4/326b2f7e-18b7-407d-a7c6-3fc641c59339.gif?v=345626351ddc8fa12dcc3c8382ac3012)

And if I try to execute in the shell it gives an interesting result:
```
$ docker pull alpine
Using default tag: latest
5053b247d78b: Already exists
82d1e9d7ed48: Already exists
e3bd82196e98: Already exists
docker.io/library/alpine:latest
```

This was performed on a fresh install of Docker (I uninstalled and reinstalled and had to reconfigure it) on macOS 13.5 Beta (22G5048d).
